server/src/workspace/workspace-resolver-builder/factories/update-many-resolver.factory.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+
+import {
+  Resolver,
+  UpdateManyResolverArgs,
+} from 'src/workspace/workspace-resolver-builder/interfaces/workspace-resolvers-builder.interface';
+import { WorkspaceSchemaBuilderContext } from 'src/workspace/workspace-schema-builder/interfaces/workspace-schema-builder-context.interface';
+import { WorkspaceResolverBuilderFactoryInterface } from 'src/workspace/workspace-resolver-builder/interfaces/workspace-resolver-builder-factory.interface';
+
+import { WorkspaceQueryRunnerService } from 'src/workspace/workspace-query-runner/workspace-query-runner.service';
+
+@Injectable()
+export class UpdateManyResolverFactory
+  implements WorkspaceResolverBuilderFactoryInterface
+{
+  public static methodName = 'updateMany' as const;
+
+  constructor(
+    private readonly workspaceQueryRunnerService: WorkspaceQueryRunnerService,
+  ) {}
+
+  create(
+    context: WorkspaceSchemaBuilderContext,
+  ): Resolver<UpdateManyResolverArgs> {
+    const internalContext = context;
+
+    return (_source, args, context, info) => {
+      return this.workspaceQueryRunnerService.updateMany(args, {
+        targetTableName: internalContext.targetTableName,
+        workspaceId: internalContext.workspaceId,
+        info,
+        fieldMetadataCollection: internalContext.fieldMetadataCollection,
+      });
+    };
```

Comment:
For some reason the resolver's return type defaults to the type of the record, for example for the `companies` collection, it is `Company` so attempting to return data in a different format has proven tricky.

### Doesn't work
![Screenshot 2023-11-29 at 11 34 35](https://github.com/twentyhq/twenty/assets/140154534/adad254b-7514-4609-aaa3-87f31da38e59)

### Work
![Screenshot 2023-11-29 at 11 38 59](https://github.com/twentyhq/twenty/assets/140154534/ba7e2d49-7b9f-48af-b3a0-3fe8c0b25dbd)
s

Reply:
@gitstart-twenty I've added few comments above, we're returning directly the records here:

```typescript
   return this.parseResult<PGGraphQLMutation<Record>>(
       result,
       targetTableName,
       'deleteFrom',
     )?.records;
```

It's why `affectedCount` is not available.
To see available fields in the introspection, just put the token in the `Headers` section of `Yoga GraphiQL` and hit the refresh button at bottom left.
This way the red underline should disappear and autocompletion will be proposed.